quinn-proto/src/connection.rs

```diff
@@ -807,14 +790,21 @@ impl Connection {
                             "discarding possible duplicate packet {packet}",
                             packet = number
                         );
+                        if self.check_for_stateless_reset(mux, &packet) {
+                            Err(ConnectionError::StatelessReset)
+                        } else {
+                            return;
+                        }
+                    } else {
                         return;
                     }
+                } else {
+                    if !was_closed {
+                        self.on_packet_authenticated(mux, now, number);
+                    }
+                    let state = self.state.take().unwrap();
+                    self.handle_connected_inner(mux, now, number, packet, state)
```

Comment:
This block should be executed regardless of whether `number.is_some()`.

Reply:
I think the flow is correct now, please check?